korangar/src/loaders/model/mod.rs

```diff
@@ -359,60 +385,122 @@ impl ModelLoader {
+            false => vec![model_data.root_node_name.clone().unwrap()],
+        };
 
-        let (root_node_position, root_node) = model_data
-            .nodes
+        let root_info: Vec<(usize, &NodeData)> = root_node_names
             .iter()
-            .enumerate()
-            .find(|(_, node_data)| &node_data.node_name == root_node_name)
-            .expect("failed to find main node");
+            .map(|node_name| {
+                let (root_node_position, root_node) = model_data
+                    .nodes
+                    .iter()
+                    .enumerate()
+                    .find(|(_, node_data)| node_data.node_name == *node_name)
+                    .expect("failed to find main node");
+                (root_node_position, root_node)
+            })
+            .collect();
 
         let mut processed_node_indices = vec![false; model_data.nodes.len()];
-        processed_node_indices[root_node_position] = true;
 
         let mut native_model_vertices = Vec::<NativeModelVertex>::new();
 
-        let mut bounding_box = AABB::uninitialized();
-        let root_node = Self::process_node_mesh(
-            root_node,
-            &model_data.nodes,
-            &mut processed_node_indices,
-            vertex_offset,
-            &mut native_model_vertices,
-            &texture_mapping,
-            &Matrix4::identity(),
-            &mut bounding_box,
-            reverse_order,
-            model_data.shade_type == 2,
-            model_data.animation_length,
-        );
-
-        let mut root_nodes = vec![root_node];
-        let is_static = root_nodes.iter().all(Self::is_static);
+        let (mut root_nodes, mut bounding_boxes): (Vec<Node>, Vec<AABB>) = root_info
+            .into_iter()
+            .map(|(root_node_position, root_node)| {
+                processed_node_indices[root_node_position] = true;
+
+                let mut bounding_box = AABB::uninitialized();
+                let root_node = Self::process_node_mesh(
+                    version,
+                    root_node,
+                    &model_data.nodes,
+                    &mut processed_node_indices,
+                    vertex_offset,
+                    &mut native_model_vertices,
+                    &texture_mapping,
+                    &Matrix4::identity(),
+                    &mut bounding_box,
+                    reverse_order ^ version.equals_or_above(2, 2),
+                    model_data.shade_type == 2,
+                    model_data.animation_length,
+                );
+                (root_node, bounding_box)
+            })
+            .unzip();
 
+        drop(texture_mapping);
+        let is_static = match version.equals_or_above(2, 2) {
+            // TODO: Need to add the dynamic RSM2.
+            true => true,
+            false => root_nodes.iter().all(Self::is_static),
+        };
+        let bounding_box = bounding_boxes
```

Comment:
Instead of doing it this way, just pull the `let mut bounding_box = AABB::uninitialized();` outside of the loop. The bounding box get's extended inside the `process_node_mesh()` anyhow.